Algorithm/QCAlgorithm.cs

```diff
@@ -162,6 +172,16 @@ public QCAlgorithm()
             _historyRequestFactory = new HistoryRequestFactory(this);
 
             _orderBasedInsightGenerator = new OrderBasedInsightGenerator();
+
+            // Framework
+            _securityValuesProvider = new AlgorithmSecurityValuesProvider(this);
+
+            // set model defaults, universe selection set via PostInitialize
+            SetAlpha(new NullAlphaModel());
+            SetPortfolioConstruction(new NullPortfolioConstructionModel());
```

Comment:
Are _all_ algorithms now considered framework -- or are we detecting `IsFramworkAlgorithm` via some mechanism?

Reply:
Exactly, all algorithms will be considered framework. In this PR I'm adding some boolean checks for backward compatibility